TLM/TLM/UI/SubTools/LaneConnectorTool.cs

```diff
@@ -283,24 +275,54 @@ private class LaneEnd {
                 }
             }
 
-            for (int cacheIndex = CachedVisibleNodeIds.Size - 1; cacheIndex >= 0; cacheIndex--) {
-                var nodeId = CachedVisibleNodeIds.Values[cacheIndex];
+            // render overlays in order. later overlays are draw over the older ones:
+
+            if (!viewOnly) {
+                RenderNodeCircles(cameraInfo);
+            }
+
+            RenderLaneCurves(cameraInfo);
 
-                bool hasMarkers = currentLaneEnds.TryGetValue((ushort)nodeId, out List<LaneEnd> laneEnds);
-                bool isNodeVisible = MainTool.IsNodeVisible(nodeId);
-                if (!viewOnly && (GetSelectionMode() == SelectionMode.None)) {
-                    Highlight.DrawNodeCircle(
+            if (!viewOnly) {
+                RenderLaneOverlays(cameraInfo);
+                RenderFloatingLaneCurve(cameraInfo);
+            }
+        }
+
+        private void RenderNodeCircles(RenderManager.CameraInfo cameraInfo) {
+            if (GetSelectionMode() == SelectionMode.None) {
+                for (int cacheIndex = CachedVisibleNodeIds.Size - 1; cacheIndex >= 0; cacheIndex--) {
+                    var nodeId = CachedVisibleNodeIds.Values[cacheIndex];
+                    bool isNodeVisible = MainTool.IsNodeVisible(nodeId);
+                    if (nodeId == HoveredNodeId && isNodeVisible) {
+                        Highlight.DrawNodeCircle(
+                            cameraInfo: cameraInfo,
+                            nodeId: this.HoveredNodeId,
+                            color: this.MainTool.GetToolColor(warning: Input.GetMouseButton(0), error: false),
+                            alpha: true,
+                            overrideRenderLimits: true);
+                    } else {
+                        Highlight.DrawNodeCircle(
                         cameraInfo: cameraInfo,
-                        nodeId: (ushort)nodeId,
+                        nodeId: nodeId,
                         color: isNodeVisible ? DefaultLaneEndColor : DefaultDisabledLaneEndColor,
                         alpha: true);
+                    }
                 }
+            }
+        }
+
+        private void RenderLaneCurves(RenderManager.CameraInfo cameraInfo) {
+            for (int cacheIndex = CachedVisibleNodeIds.Size - 1; cacheIndex >= 0; cacheIndex--) {
```

Comment:
depends on mode but it can loop twice. why do you ask? performance?
```
// before:
for each element in array:
	do x
	if node selection mode:
		do y
	

// after:
for each element in array:
	do x

if node selection mode:
	for each element in array:	
		do y
```
considering how slow `do x` and `do y` is, I think the array loops influence is negligible.